sdk/cosmos/azure-cosmos/azure/cosmos/aio/container.py

```diff
@@ -0,0 +1,714 @@
+        return cast('ScriptsProxy', self._scripts)
+
+    def _get_document_link(self, item_or_link):
+        # type: (Union[Dict[str, Any], str]) -> str
+        if isinstance(item_or_link, str):
+            return u"{}/docs/{}".format(self.container_link, item_or_link)
+        return item_or_link["_self"]
+
+    def _get_conflict_link(self, conflict_or_link):
+        # type: (Union[Dict[str, Any], str]) -> str
+        if isinstance(conflict_or_link, str):
+            return u"{}/conflicts/{}".format(self.container_link, conflict_or_link)
+        return conflict_or_link["_self"]
+
+    async def _set_partition_key(self, partition_key):
+        if partition_key == NonePartitionKeyValue:
+            return CosmosClientConnection._return_undefined_or_empty_partition_key(await self.is_system_key)
+        return partition_key
+
+    @distributed_trace_async
+    async def read(
+        self,
+        populate_partition_key_range_statistics=None,  # type: Optional[bool]
+        populate_quota_info=None,  # type: Optional[bool]
+        **kwargs  # type: Any
+    ):
+        # type: (...) -> Dict[str, Any]
+        """Read the container properties.
+
+        :param populate_partition_key_range_statistics: Enable returning partition key
+            range statistics in response headers.
+        :param populate_quota_info: Enable returning collection storage quota information in response headers.
+        :keyword str session_token: Token for use with Session consistency.
+        :keyword dict[str,str] initial_headers: Initial headers to be sent as part of the request.
+        :keyword Callable response_hook: A callable invoked with the response metadata.
+        :raises ~azure.cosmos.exceptions.CosmosHttpResponseError: Raised if the container couldn't be retrieved.
+            This includes if the container does not exist.
+        :returns: Dict representing the retrieved container.
+        :rtype: dict[str, Any]
+        """
+        request_options = build_options(kwargs)
+        response_hook = kwargs.pop('response_hook', None)
+        if populate_partition_key_range_statistics is not None:
+            request_options["populatePartitionKeyRangeStatistics"] = populate_partition_key_range_statistics
+        if populate_quota_info is not None:
+            request_options["populateQuotaInfo"] = populate_quota_info
+
+        collection_link = self.container_link
+        self._properties = await self.client_connection.ReadContainer(
+            collection_link, options=request_options, **kwargs
+        )
+
+        if response_hook:
+            response_hook(self.client_connection.last_response_headers, self._properties)
+
+        return cast('Dict[str, Any]', self._properties)
+
+    @distributed_trace_async
+    async def create_item(
+        self,
+        body,  # type: Dict[str, Any]
+        pre_trigger_include=None,  # type: Optional[str]
+        post_trigger_include=None,  # type: Optional[str]
```

Comment:
after the first preview, we should consider making commonly used parameters as explicit parameters and make non commonly used one as non-explicit in the kwargs